src/Renderer/ThreeExtended/GlobeControls.js

```diff
@@ -1506,6 +1606,130 @@ GlobeControls.prototype.isAnimationEnabled = function isAnimationEnabled() {
 };
 
+
+/**
+ * Returns the actual zoom level. The level will always be between the [getMinZoomLevel(), getMaxZoomLevel()].
+ * @constructor
+ * @return     {number}  The zoom level.
+ */
+GlobeControls.prototype.getZoomLevel = function getZoomLevel() {
+    return computeTileZoomFromDistanceCamera(this.getRange());
+};
+
+/**
+ * Gets the current zoom level, which is an index in the logical scales predefined for the application.
+ * The higher the level, the closer to the ground.
+ * The level is always in the [getMinZoomLevel(), getMaxZoomLevel()] range.
+ * @constructor
+ * @param      {number}  zoom    The zoom
+ * @param      {boolean}  isAnimated  Indicates if animated
+ * @return     {Promise}
+ */
+GlobeControls.prototype.setZoomLevel = function setZoomLevel(zoom, isAnimated) {
+    isAnimated = isAnimated === undefined ? this.isAnimationEnabled() : isAnimated;
+    const range = computeDistanceCameraFromTileZoom(zoom);
+    return this.setRange(range, isAnimated);
+};
+
+/**
+ * Return the current zoom scale at the central point of the view.
+ * This function compute the scale of a map
+ * @constructor
+ * @param      {number}  pitch   Screen pitch, in millimeters ; 0.28 by default
+ * @return     {number}  The zoom scale.
+ */
+GlobeControls.prototype.getScale = function getScale(pitch) {
+    // TODO: Why error div size height in Chrome?
+    // Screen pitch, in millimeters
+    pitch = (pitch || 0.28) / 1000;
+    const FOV = sizeRendering.FOV / 180 * Math.PI * 0.5;
+    // projection one unit on screen
+    const unitProjection = sizeRendering.height / (2 * this.getRange() * Math.tan(FOV));
+    return pitch * unitProjection;
+};
+
+/**
+ * Changes the zoom level of the central point of screen so that screen acts as a map with a specified scale.
+ *  The view flies to the desired zoom scale;
+ * @param      {number}  scale  The scale
+ * @param      {number}  pitch  The pitch
+ * @param      {boolean}  isAnimated  Indicates if animated
+ * @return     {Promise}
+ */
+GlobeControls.prototype.setScale = function setScale(scale, pitch, isAnimated) {
+    isAnimated = isAnimated === undefined ? this.isAnimationEnabled() : isAnimated;
+    const range = getRangeFromScale(scale);
+    return this.setRange(range, isAnimated);
+};
+
+/**
+ * Changes the center of the scene on screen to the specified coordinates.
+ * <iframe width="100%" height="400" src="//jsfiddle.net/iTownsIGN/x06yhbq6/embedded/" allowfullscreen="allowfullscreen" frameborder="0"></iframe>
+ * @param {Object} coordinates - The globe coordinates in EPSG_4326 projection to aim to
+ * @param {number} coordinates.latitude
+ * @param {number} coordinates.longitude
+ * @param {number} coordinates.range
+ * @param {boolean}  isAnimated - if the movement should be animated
+ * @return {Promise} A promise that resolves when the next 'globe-loaded' event fires.
+ */
+GlobeControls.prototype.setCameraTargetGeoPosition = function setCameraTargetGeoPosition(coordinates, isAnimated) {
```

Comment:
Doesn't do anything when called in index.html when animation is set to true. (works when animation is set to false)

Works fine in console.